plugins/woocommerce/src/Internal/Orders/SourceAttributionBlocksController.php

```diff
@@ -0,0 +1,245 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\Orders;
+
+use Automattic\Jetpack\Constants;
+use Automattic\WooCommerce\Internal\Features\FeaturesController;
+use Automattic\WooCommerce\Internal\Orders\SourceAttributionController;
+use Automattic\WooCommerce\Internal\RegisterHooksInterface;
+use Automattic\WooCommerce\StoreApi\Schemas\ExtendSchema;
+use Automattic\WooCommerce\StoreApi\Schemas\V1\CheckoutSchema;
+use Automattic\WooCommerce\Internal\Traits\ScriptDebug;
+use WP_Error;
+
+/**
+ * Class SourceAttributionBlocksController
+ *
+ * @since x.x.x
+ */
+class SourceAttributionBlocksController implements RegisterHooksInterface {
+
+	use ScriptDebug;
+
+	/**
+	 * Instance of the features controller.
+	 *
+	 * @var FeaturesController
+	 */
+	private $features_controller;
+
+	/**
+	 * ExtendSchema instance.
+	 *
+	 * @var ExtendSchema
+	 */
+	private $extend_schema;
+
+	/**
+	 * Instance of the source attribution controller.
+	 *
+	 * @var SourceAttributionController
+	 */
+	private $source_attribution_controller;
+
+	/**
+	 * Bind dependencies on init.
+	 *
+	 * @internal
+	 *
+	 * @param ExtendSchema                $extend_schema                 ExtendSchema instance.
+	 * @param FeaturesController          $features_controller           Features controller.
+	 * @param SourceAttributionController $source_attribution_controller Instance of the source attribution controller.
+	 */
+	final public function init(
+
+		/*
+		AssetApi $asset_api,
+		*/
+		ExtendSchema $extend_schema,
+		FeaturesController $features_controller,
+		SourceAttributionController $source_attribution_controller
+	) {
+		// $this->asset_api                     = $asset_api;
+		$this->extend_schema                 = $extend_schema;
+		$this->features_controller           = $features_controller;
+		$this->source_attribution_controller = $source_attribution_controller;
+	}
+
+	/**
+	 * Hook into WP.
+	 *
+	 * @since x.x.x
+	 * @return void
+	 */
+	public function register() {
+		// Bail if the feature is not enabled.
+		if ( ! $this->features_controller->feature_is_enabled( 'order_source_attribution' ) ) {
+			return;
+		}
+
+		$this->extend_api();
+
+		// Bail early on admin requests to avoid asset registration.
+		if ( is_admin() ) {
+			return;
+		}
+
+		add_action(
+			'init',
+			function() {
+				$this->register_assets();
+			}
+		);
+
+		add_action(
+			'wp_enqueue_scripts',
+			function() {
+				$this->enqueue_scripts();
+			}
+		);
+	}
+
+	/**
+	 * Check if WCCom_Cookie_Terms is available.
+	 *
+	 * @return bool
+	 */
+	protected function is_wccom_cookie_terms_available() {
+		return class_exists( WCCom_Cookie_Terms::class );
+	}
```

Comment:
We can probably remove this method (https://github.com/woocommerce/woocommerce/pull/39701#issuecomment-1820274517)
```suggestion
```